src/EnergyPlus/General.hh

```diff
@@ -252,6 +252,175 @@ namespace General {
         XRes = XTemp;
     }
 
+    // A second version that does not require a payload -- use lambdas
```

Comment:
This is a second version of the SolveRoot template which accepts a function that only takes a single X value, and it does not take a payload argument anymore.  Once we aren't using the payload version anymore, it will be deleted and this will be the only one.  Possibly within this branch itself, given how quickly the conversion is going.